src/components/dateTimePicker/index.js

```diff
@@ -140,8 +148,14 @@ class DateTimePicker extends BaseComponent {
 
   getStringValue = () => {
     const {value} = this.state;
-    const {mode, dateFormat, timeFormat} = this.getThemeProps();
+    const {mode, dateFormat, timeFormat, dateFormatter, timeFormatter} = this.getThemeProps();
     if (value) {
+      if (dateFormatter && mode === MODES.DATE) {
```

Comment:
I would use a switch case here and include all logic in each case to make the code more readable. What do you think?

Reply:
I changed to switch case - not sure what is more readable..
Let me know if you have an idea for a better logic here :)